gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calculation
+
+import (
+	"path"
+	"sort"
+	"strings"
+)
+
+// CoverageList is a collection and summary over multiple file Coverage objects
+type CoverageList struct {
+	*coverage
+	Group             []coverage
+	CoverageThreshold int
+}
+
+// CovList constructs new (file) Group Coverage
+func newCoverageList(name string, covThresholdInt int) *CoverageList {
+	return &CoverageList{
+		coverage:          &coverage{Name: name},
+		Group:             []coverage{},
+		CoverageThreshold: covThresholdInt,
+	}
+}
+
+func (covList *CoverageList) ratio() float32 {
+	covList.Summarize()
+	return covList.coverage.ratio()
+}
+
+// Summarize summarizes all items in the Group and stores the result
+func (covList *CoverageList) Summarize() {
+	for _, item := range covList.Group {
+		covList.nCoveredStmts += item.nCoveredStmts
+		covList.nAllStmts += item.nAllStmts
+	}
+}
+
+func (covList *CoverageList) append(c *coverage) {
```

Comment:
This should probably be exported (`Append`), given it manipulates an exported property. That said, given the exported property is accessed directly from outside `CoverageList`, I'm not sure this really serves any purpose as-is anyway?

Reply:
why should a method manipulating an exported property also be exported?

Reply:
do we need this method? I can do `append(covList.Group, *c)` in the caller itself to append to group since Group is exported.